packages/slice-machine/src/modules/selectedSlice/sagas.ts

```diff
@@ -0,0 +1,328 @@
+        error: response.data.err,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        status: response.status,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        message: response.data.reason,
+      });
+    }
+    setData({
+      loading: false,
+      done: true,
+      error: null,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+      warning: !!response.data.warning,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+      status: response.status,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+      message:
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        response.data.warning ||
+        "Models & mocks have been generated successfully!",
+    });
+    yield put(
+      saveSliceCreator.success({
+        extendedComponent,
+      })
+    );
+  } catch (e) {
+    yield put(
+      openToasterCreator({
+        message: "Internal Error: Models & mocks not generated",
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+}
+
+export function* pushSliceSaga({
+  payload,
+}: ReturnType<typeof pushSliceCreator.request>) {
+  const { extendedComponent, onPush } = payload;
+  try {
+    onPush({
+      imageLoading: true,
+      loading: true,
+      done: false,
+      error: null,
+      status: null,
+    });
+    const response = (yield call(
+      pushSliceAxios,
+      extendedComponent.component
+    )) as SagaReturnType<typeof pushSliceAxios>;
+    if (response.status > 209) {
+      return onPush({
+        imageLoading: false,
+        loading: false,
+        done: true,
+        error: null,
+        status: response.status,
+      });
+    }
+    onPush({
+      imageLoading: false,
+      loading: false,
+      done: true,
+      error: null,
+      status: response.status,
+    });
+    yield put(
+      pushSliceCreator.success({
+        extendedComponent,
+      })
+    );
+    yield put(
+      openToasterCreator({
+        message: "Model was correctly saved to Prismic!",
+        type: ToasterType.SUCCESS,
+      })
+    );
+  } catch (e) {
+    yield put(
+      openToasterCreator({
+        message: "Internal Error: Slice was not pushed",
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+}
+
+export function* renameSliceSaga({
+  payload,
+}: ReturnType<typeof renameSliceCreator.request>) {
+  try {
+    yield call(
+      renameSlice,
+      payload.sliceId,
+      payload.newSliceName,
+      payload.libName
+    );
+    //TODO: avoid refetch of state. Either manually update all the changes within the reducer,
```

Comment:
ah I have changed this in my other PR sorry! it's slightly different now.
If it's annoying to rebase and change dw about it just delete it and I will do the saga change myself :)

Reply:
changed np

Reply:
thanks!